x-pack/plugins/security_solution/server/endpoint/utils/create_internal_readonly_so_client.ts

```diff
@@ -51,7 +52,7 @@ export const createInternalReadonlySoClient = (
       methodName: SavedObjectsClientContractKeys,
       receiver: unknown
     ): unknown {
-      if (RESTRICTED_METHODS.includes(methodName)) {
+      if (!allowList.has(methodName) && RESTRICTED_METHODS.includes(methodName)) {
```

Comment:
I'm not crazy about this change. The goal of this function it to create a Read Only internal SO client. This change overrides that.  I'm thinking that the changes you are making with regards to transforms install/restart need to have its own service client that creates its own internal "regular" SO client that can perform updates.

Reply:
Agreed that the creation of the internal SO client you need should not be through the `InternalReadonlySoClient`. 

You are creating specifically a `DangerousInternalSOClient`. Copy what you need from here, put it in a new place for a dangerousISOC creator, and put it in the most-specific place near the task logic. Away from generic importables/commons. Giant warning comment that it should *NEVER* be used in a user request / HTTP handler call stack.

Reply:
👀  `InternalReadonlySoClient` 👀 ... 👀 `Readonly`👀 . I uh...should probably read next time.